File: Git과 Python/codes/python_basic/lotto.js
const _ = require('lodash');
// 1부터 45까지의 숫자 중, 6개를 '비복원 추출'한다.
// > []
// let lotto = [];
// let num = [];
// lotto = _.range(1,45);
// lotto = _.shuffle(lotto);
// for (let i = 0; i < 6; i++) {
//     num.push(lotto.pop());
// }
// sampleSize로 편하게 가능 (_.sampleSize(lotto, 6))
// var bubbleSort = function (array) {
//     var length = array.length-1;
//     var i, j, temp;
//     for (i = 0; i < length; i++) { 
//         for (j = 0; j < length - i; j++) {
//             if (array[j] > array[j + 1]) { 
//                 temp = array[j]; 
//                 array[j] = array[j + 1];
//                 array[j + 1] = temp;
//             }
//         }
//     }
//     return array;
// };
// num = bubbleSort(num);

// console.log(_.sortBy(_.sampleSize(_.range(1, 45), 6)));

// const winner = [1, 15, 19, 23, 28, 42];
// const bonus = [7];
// let lotto = _.sortBy(_.sampleSize(_.range(1, 45), 6));
// let lottobonus = _.xor([7], _.sortBy(_.sampleSize(_.range(1, 45), 6)));
// let match = (12 - _.xor([1, 15, 19, 23, 28, 42], _.sortBy(_.sampleSize(_.range(1, 45), 6))).length) / 2;
// console.log(match);
// _.intersection 활용하자
switch ((12 - _.xor([1, 15, 19, 23, 28, 42], _.sortBy(_.sampleSize(_.range(1, 45), 6))).length) / 2) {
    case 0:
        console.log('꽝!');
        break;
    case 1:
        console.log('꽝!');
        break;
    case 2:
        console.log('꽝!');
        break;
    case 3:
        console.log('5등 당첨!');
        break;
    case 4:
        console.log('4등 당첨!');
        break;
    case 5:
        if(_.xor([7], _.sortBy(_.sampleSize(_.range(1, 45), 6))).length===5) console.log('2등 당첨!')
        console.log('3등 당첨!');
        break;
    case 6:
        console.log('1등 당첨!');
        break;
    default:
        console.log('뭐지?')
}

File: git과python/python_basic/lotto.py
import random

def lotto():
    winner = [1, 15, 19, 23, 28, 42]
    bonus = [6]
    # random.sample([배열], 뽑을 개수)
    lotto = sorted(random.sample(range(1, 46), 6))

    # set(배열) → 집합(중복원소 제거, 집합 연산자)
    set(lotto)

    # python은 .length 대신 len함수가 있다.
    num = len(set(lotto) & set(winner))
    # 조건문을 사용할 때 python은  중괄호가 없기 때문에 :를 붙여주고 다음줄에 인덴팅을 해준다.
    if num==3:
        rank = "5등 당첨!"
    elif num==4:
        rank = "4등 당첨!"
    elif num==5:
        rank = "3등 당첨!"
    elif num==5 & len(set(lotto) & set(bonus))==1:
        rank = "2등 당첨!"
    elif num==6:
        rank = "1등 당첨!"
    else:
        rank = "꽝!"

    print(f'{num}개 일치! {rank}')
n=0
while n<10000:
    lotto()
    n += 1
